int64  i
int64  j
---
int64   w
int64   a
int64   s
int64   d
